packages/wallet-sdk/src/components/LinkFlow/LinkFlow.tsx

```diff
@@ -51,40 +41,35 @@ export class LinkFlow {
     this.sessionSecret = options.sessionSecret;
     this.linkAPIUrl = options.linkAPIUrl;
     this.isParentConnection = options.isParentConnection;
-    this.connected$ = options.connected$;
-    this.chainId$ = options.chainId$;
+    this.connected = options.connected;
+    this.chainId = options.chainId;
   }
 
   public attach(el: Element): void {
     this.root = document.createElement('div');
     this.root.className = '-cbwsdk-link-flow-root';
     el.appendChild(this.root);
     this.render();
+  }
 
-    this.subscriptions.add(
-      this.connected$.subscribe((v) => {
-        if (this.isConnected !== v) {
-          this.isConnected = v;
-          this.render();
-        }
-      })
-    );
+  public setConnected(v: boolean) {
```

Comment:
checking whether `attached` (and introducing local var to store it) is not necessary,
since `render()` checks whether `this.root` exists.